builder/chroots.go

```diff
@@ -80,6 +82,459 @@ func readBuildChrootsConfig(path string) (*buildChrootsConfig, error) {
+	// version to be different. Use SOURCE_DATE_EPOCH if available?
+	versionstamp := fmt.Sprint(time.Now().Unix())
+	return ioutil.WriteFile(filepath.Join(clearDir, "versionstamp"), []byte(versionstamp), 0644)
+}
+
+func initRPMDB(chrootDir string) error {
+	err := os.MkdirAll(filepath.Join(chrootDir, "var/lib/rpm"), 0755)
+	if err != nil {
+		return err
+	}
+
+	return helpers.RunCommandSilent(
+		"rpm",
+		"--root", chrootDir,
+		"--initdb",
+	)
+}
+
+func buildOsCore(packagerCmd []string, chrootDir, version string) error {
+	err := initRPMDB(chrootDir)
+	if err != nil {
+		return err
+	}
+
+	if err := installFilesystem(packagerCmd, chrootDir); err != nil {
+		return err
+	}
+
+	if err := createClearDir(chrootDir, version); err != nil {
+		return err
+	}
+
+	if err := fixOSRelease(filepath.Join(chrootDir, "usr/lib/os-release"), version); err != nil {
+		return errors.Wrap(err, "couldn't fix os-release file")
+	}
+
+	if err := createVersionsFile(filepath.Dir(chrootDir), packagerCmd); err != nil {
+		return errors.Wrapf(err, "couldn't create the versions file")
+	}
+
+	return nil
+}
+
+func genUpdateBundleSpecialFiles(chrootDir string, cfg *buildChrootsConfig, b *Builder) error {
+	swupdDir := filepath.Join(chrootDir, "usr/share/defaults/swupd")
+	if err := os.MkdirAll(swupdDir, 0755); err != nil {
+		return err
+	}
+	cURLBytes := []byte(cfg.ContentURL)
+	if err := ioutil.WriteFile(filepath.Join(swupdDir, "contenturl"), cURLBytes, 0644); err != nil {
+		return err
+	}
+	if err := ioutil.WriteFile(filepath.Join(swupdDir, "versionurl"), cURLBytes, 0644); err != nil {
+		return err
+	}
+
+	// Only copy the certificate into the mix if it exists
+	if _, err := os.Stat(b.Cert); err == nil {
+		certdir := filepath.Join(chrootDir, "/usr/share/clear/update-ca")
+		err = os.MkdirAll(certdir, 0755)
+		if err != nil {
+			return err
+		}
+		chrootcert := filepath.Join(certdir, "Swupd_Root.pem")
+		err = helpers.CopyFile(chrootcert, b.Cert)
+		if err != nil {
+			return err
+		}
+	}
+
+	return ioutil.WriteFile(filepath.Join(swupdDir, "format"), []byte(b.Format), 0644)
+}
+
+func installBundleToFull(packagerCmd []string, chrootVersionDir string, bundle *bundle) error {
+	baseDir := filepath.Join(chrootVersionDir, "full")
+	args := merge(packagerCmd, "--installroot="+baseDir, "install")
+	args = append(args, bundle.AllPackages...)
+	err := helpers.RunCommandSilent(args[0], args[1:]...)
+	if err != nil {
+		return err
+	}
+
+	bundleDir := filepath.Join(baseDir, "usr/share/clear/bundles")
+	err = os.MkdirAll(filepath.Join(bundleDir), 0755)
+	if err != nil {
+		return err
+	}
+
+	return ioutil.WriteFile(filepath.Join(bundleDir, bundle.Name), nil, 0644)
+}
+
+func buildFullChroot(cfg *buildChrootsConfig, b *Builder, set *bundleSet, packagerCmd []string, chrootVersionDir, version string) error {
+	fmt.Println("Installing all bundles to full chroot")
+	totalBundles := len(*set)
+	i := 0
+	for _, bundle := range *set {
```

Comment:
`Optionally` next time could save a line and just do for i, bundle := ... etc 

:smile: :100:

Reply:
Actually saves 2 lines, the `i := 0` and the `i++`

Reply:
doh, I know how to write Go I swear. Done.